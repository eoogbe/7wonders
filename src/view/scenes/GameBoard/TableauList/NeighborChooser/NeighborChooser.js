/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import './NeighborChooser.css';

const NEIGHBORS = [
  { costType: 'leftCost', text: 'Left neighbor' },
  { costType: 'rightCost', text: 'Right neighbor' },
];

const renderButton = ({ chooseNeighbor }) => ({ costType, text }) => (
  <li key={costType}>
    <button
      className="primary-button"
      type="button"
      onClick={() => chooseNeighbor(costType)}
    >
      {text}
    </button>
  </li>
);

const NeighborChooser = ({ chooseNeighbor }) => (
  <section className="NeighborChooser">
    <h2 className="NeighborChooser-heading">
      Choose a neighbor to trade with:
    </h2>
    <ul className="NeighborChooser-buttonPanel">
      {NEIGHBORS.map(renderButton({ chooseNeighbor }))}
    </ul>
  </section>
);

export default NeighborChooser;
