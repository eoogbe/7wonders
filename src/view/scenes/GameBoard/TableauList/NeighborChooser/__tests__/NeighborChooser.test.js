/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NeighborChooser from '../NeighborChooser';

it('renders without crashing', () => {
  const tree = renderer.create(<NeighborChooser />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('chooses the left neighbor', () => {
  const chooseNeighbor = jest.fn();
  const tree = shallow(<NeighborChooser chooseNeighbor={chooseNeighbor} />);
  tree.find('button').first().simulate('click');
  expect(chooseNeighbor).toHaveBeenCalledWith('leftCost');
});

it('chooses the right neighbor', () => {
  const chooseNeighbor = jest.fn();
  const tree = shallow(<NeighborChooser chooseNeighbor={chooseNeighbor} />);
  tree.find('button').last().simulate('click');
  expect(chooseNeighbor).toHaveBeenCalledWith('rightCost');
});
