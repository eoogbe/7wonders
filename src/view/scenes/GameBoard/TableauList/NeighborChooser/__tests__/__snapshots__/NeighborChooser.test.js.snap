// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<section
  className="NeighborChooser"
>
  <h2
    className="NeighborChooser-heading"
  >
    Choose a neighbor to trade with:
  </h2>
  <ul
    className="NeighborChooser-buttonPanel"
  >
    <li>
      <button
        className="primary-button"
        onClick={[Function]}
        type="button"
      >
        Left neighbor
      </button>
    </li>
    <li>
      <button
        className="primary-button"
        onClick={[Function]}
        type="button"
      >
        Right neighbor
      </button>
    </li>
  </ul>
</section>
`;
